src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreWorker.cs

```diff
@@ -223,9 +223,8 @@ private void Reset(bool isDisposing = false)
                     {
                         // Do not block VS forever
                         // After the specified delay the task will disjoin.
-                        await _backgroundJobRunner.GetValueAsync().WithTimeout(TimeSpan.FromSeconds(60));
-                    },
-                    JoinableTaskCreationOptions.LongRunning);
+                        await _backgroundJobRunner.GetValueAsync().WithTimeout(TimeSpan.FromSeconds(5));
```

Comment:
Do we know why the cancellation token does not get propagated for that long?

Reply:
because cancellation token is passed through al the restore code path properly like in this case, it hasn't been passed through while generating the assets file so it goes through all the packages dependencies to generate libraries section for assets file. I've logged an tracking issue to improve this - https://github.com/NuGet/Home/issues/7564